### ISSUE TITLE
MangoApp config file location

### ISSUE THREAD

**Issue by r-vd-woude** [flightlessmango/MangoHud#944]
Maybe I'm not looking correctly, but I can find no documentation on how to configure MangoApp when running with Gamescope on a normal desktop pc.

Would like to be able to do that, but maybe it is not possible.

Thanks for the hard work on MangoHud, it is an amazing piece of software!

**Comment by flightlessmango** [comment]
It shouldn't be any different from mangohud, it uses the same locations etc

**Comment by r-vd-woude** [comment]
Thanks for the quick reply!
My config for MangoHud is located in ~/.config/MangoHud

MangoHud does not seem to use the config present there.

Will investigate a bit further.

**Comment by r-vd-woude** [comment]
First, MangoApp does look for _mangoapp.conf_ not _MangoHud.conf_, that was my mistake.

Secondly, my _MangoHud.conf_ contained some mistakes, and would therefore not load properly inside Gamescope.

This issue can be closed.

**Comment by zoobporsor** [comment]
I am having the same problem. When I use mangoapp with gamescope, it uses the default config instead of my custom one. I tried creating   ~/.config/mangoapp.conf with the same lines as my custom one, but it still doesnt take effect

**Comment by ConorVernon** [comment]
I have also tried copying my `MangoHud.conf` file as `.config/MangoHud/mangoapp.conf` but observed no changes to the behaviour of mangoapp in gamescope. Tested by running `gamescope --mangoapp glxgears` using a config with an fps limit far below my refresh rate (240hz, 60fps limit) where the result was always glxgears running at 240fps.

However, saving a change to that file does appear to reload it, which I can tell because it's currently throwing an error at me when it does so: `[MANGOHUD] [error] [overlay_params.cpp:695] Unknown option 'Shift_L+F2'` - which appears to happen regardless of the setting I use. 

This is occuring on the up-to-date openSUSE packages of Mangohud/Mangoapp (0.7.2) and gamescope (3.14.26). I may experiment with building from source soon to see if this still behaves the same way.

**Comment by flightlessmango** [comment]
@ConorVernon gamescope creates it's own mangohud config file, you need to override it with `MANGOHUD_CONFIGFILE` or use a later gamescope version where it's fixed

**Comment by ConorVernon** [comment]
> @ConorVernon gamescope creates it's own mangohud config file, you need to override it with `MANGOHUD_CONFIGFILE` or use a later gamescope version where it's fixed

I had seen mention of this, particularly on [this Steam Deck reddit thread](https://www.reddit.com/r/SteamDeck/comments/v9mxtb/customize_perf_overlay_mangohud_presets/), but none of these options worked for me.

At least, not by themselves! After I tested gamescope built from source and had the same issue, I tinkered some more and found I need to set two environment variables at once:

1. `MANGOHUD_CONFIGFILE=<path-to-config.conf>`
2. `MANGOHUD_CONFIG=read_cfg`

The combo of loading the file and explicitly asking MangoHud to read it results in the config being loaded properly.
